cylc/flow/task_pool.py

```diff
@@ -1677,25 +1675,33 @@ def match_taskdefs(
         """
         n_warnings = 0
         task_items: Dict[Tuple[str, 'PointBase'], 'TaskDef'] = {}
-        for item in items:
-            tokens = Tokens(item, relative=True)
+        for id_ in ids:
+            try:
+                tokens = Tokens(id_, relative=True)
+            except ValueError:
+                LOG.warning(f'Invalid task ID: {id_}')
```

Comment:
As with https://github.com/cylc/cylc-flow/pull/4597#issuecomment-1017699862
```suggestion
                LOG.warning(f'Invalid task ID: {id_}')
                continue
```

Reply:
Will take a look

Reply:
Yep, that's a glaring bug.